site/docs/guide/context.md

```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
 
 Hence, if you want to, you can forget about `ctx.message` and `ctx.channelPost` and `ctx.editedMessage` and so on and so forth, and just always use `ctx.msg` instead.
 
+## Probing via Has Checks
+
+The context object has a few methods that allow you to probe the contained data for certain things.
+For example, you can call `ctx.hasCommand("start")` to see if the context object contains a `/start` command.
+This is why the methods are collectively named _has checks_.
+
+::: tip Know When to Use Has Checks
+
+This is the exact same logic that is used by `bot.command("start")`.
+Note that you should usually use [filter queries](./filter-queries.md) and similar methods.
+Using has checks works best inside the [conversations plugin](../plugins/conversations.md).
+
+:::
+
+The has checks correctly narrow down the context type.
+This means that checking if a context has callback query data will tell TypeScript that the context has the field `ctx.callbackQuery.data` present.
```

Comment:
> This means that checking if a context has callback query data will tell TypeScript that the context has the field `ctx.callbackQuery.data` present.

Would you guys please explain to me what this sentence means? I don't understand the full meaning of the sentence :thinking:

Reply:
It means that the has functions are type predicates (https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) which narrow down the type of the context object.